fnmatch.go

```diff
@@ -0,0 +1,169 @@
+package editorconfig
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+func FnmatchCase(pattern, name string) bool {
+	p, _ := translate(pattern)
+	r, err := regexp.Compile(fmt.Sprintf("^%s$", p))
+
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	return r.MatchString(name)
+}
+
+func translate(pattern string) (string, error) {
+	index := 0
+	pat := []rune(pattern)
+	length := len(pat)
+
+	result := strings.Builder{}
+
+	braceLevel := 0
+	isEscaped := false
+	inBrackets := false
+
+	findLeftBrackets := regexp.MustCompile(`(^|[^\\])\{`)
+	findRightBrackets := regexp.MustCompile(`(^|[^\\])\}`)
+	matchesBraces := len(findLeftBrackets.FindAllString(pattern, -1)) == len(findRightBrackets.FindAllString(pattern, -1))
+
+	findNumericRange := regexp.MustCompile(`^([+-]?\d+)\.\.([+-]?\d+)$`)
```

Comment:
I think these static regexpes should probably be global (unnexported) variables, to prevent the performance penalty of having to compile them on every function call, instead of once in the app initialization.